tests/algorithms/test_optimizers.py

```diff
@@ -50,7 +38,6 @@ def test_optimizers_rosenbrock(optimizer, enforce_bounds_on_x1):
 
         # save to compare with later as optimization should not change the initial points
         params_init_before = [i.detach().clone for i in params_init]
-        params_init_grad_before = [i.grad.clone() for i in params_init]
```

Comment:
why is this removed?

Reply:
remark: maybe would be even better to also randomly set  i.grad before to some values that should not be touched by the optimizer.

Reply:
> why is this removed?

good question... probably happened in one of the semi-manual corrections of the initial ruff formats adding strict=False everywhere...

Reply:
I fixed it and made sure the test actually runs and doesnt always pass no matter what the optimizer does.